packages/store/addon/-private/system/core-store.ts

```diff
@@ -3386,6 +3392,12 @@ abstract class CoreStore extends Service {
       serializer && deprecateTestRegistration('serializer', '-default');
     }
 
+    deprecate('deprecate -default serializer fallback in store.serializerFor', !serializer, {
```

Comment:
let's leave a better description for folks that may not know what this method is or what it is doing